corpus/convert.py

```diff
@@ -182,3 +183,45 @@ def run(self):
                 segment.orth = line.strip()
 
         c.dump(tk.uncached_path(self.out_corpus))
+
+
+class CorpusToTextDictJob(Job):
+    """
+    Extract the Text from a Bliss corpus to fit a "{key: text}" structure (e.g. for RETURNN)
+    """
+
+    def __init__(self, corpus, segments=None):
+        """
+        :param Path corpus: bliss corpus file
+        :param Path|None segments: a segment file as optional whitelist
+        """
+        self.corpus_path = corpus
+        self.segments_file_path = segments
+
+        self.out_dictionary = self.output_path("text_dictionary.py")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        c = corpus.Corpus()
+        c.load(tk.uncached_path(self.corpus_path))
+
+        dictionary = {}
+
+        segments = None
+        if self.segments_file_path:
+            with uopen(self.segments_file_path) as f:
+                segments = set(line.decode().strip() for line in f.readlines())
```

Comment:
Isn't `for line in f` more memory efficient since it does not load the complete file at once? not 100% sure.. 
not sure if this is even really relevant.. more curiosity on my side :)